doc/tutorials/notifications.md

```diff
@@ -0,0 +1,229 @@
+# Notification Tutorial
+
+## Preface
+
+**The features described in this document are experimental or not implemented
+yet.**
+
+This document explains how notifications are implemented in Elektra and how
+they can be used by application developers.
```

Comment:
What is the difference between io lib and io binding?

Reply:
`libelektra-io` contains utility functions for creating operations/bindings and accessing their fields (e.g. `elektraIoNewFdOperation`, `elektraIoNewBinding` and `elektraIoTimerIsEnabled`) while I/O bindings contain the functions (passed to `elektraIoNewBinding`) for a specific I/O management library.

Reply:
Thanks, that should be a part of the tutorial or docu.